CONTRIBUTING.md

```diff
@@ -161,55 +166,132 @@ To minimize unnecessary updates of the local apt database running `apt-get updat
 
 ## Test Architecture
 
-Testing is based on three pilars: A) *Installation of base system*, B) *Test Cases*, and C) *Static analysis using linter*.
+Testing is based on three pillars:
+
+A) *Installation of base system*
+
+B) *Test Cases using [BATS](https://github.com/bats-core/bats-core)*
+
+C) *Static analysis using the [Shellcheck](https://www.shellcheck.net/) linter*
 
 ### Test installation
-Test installations are done continuously using Docker on a Travis Virtual Machine and by testing on actual hardware, eg. Raspberry Pi. A Docker installation can be performed by three commands. Firstly a Docker image is built where the `openhabian` code is injected (see `Dockerfile.*` for details):
+Test installations are done continuously using Docker on a Travis Virtual
+Machine and by testing on actual hardware, eg. Raspberry Pi. A Docker
+installation can be performed by three commands. Firstly a Docker image is built
+where the `openhabian` code is injected (see `Dockerfile.*` for details).
 
+To begin, first make a Docker container for your platform. An example Docker
+container build for `amd64` would look like:
 ```
-docker build --tag openhabian/openhabian-bats .
+docker build --tag openhabian/install-openhabian -f Dockerfile.amd64 .
 ```
-The openHABian scripts are using `systemd` for service management, to use `systemd` with Docker the container must be started first to ensure `systemd` gets pid 1. This is done by executing:
 
+While not a problem in the final container-less deployment, with more than one
+container running, all but one of them will have a `systemd` running with a PID
+other than 1. That will break openHABian in a number of locations when we use
+`systemctl` to start/stop/check `systemd` controlled services.
+This is why we replace the `systemctl` binary in `Dockerfile.*` with Travis
```

Comment:
"with"? 

And add whitespace after dot next line

Reply:
done